src/rest/RequestHandler.js

```diff
@@ -176,6 +176,19 @@ class RequestHandler {
       return this.execute(request);
     }
 
+    if (this.manager.client.listenerCount(API_RESPONSE)) {
+      /**
+       * Emitted after every api request has received a response.
+       * This event does not necessarily correlate to completion of the request, e.g. when hitting a rate limit.
+       * <info>This is an informational event that is emitted quite frequently,
+       * it is highly recommended to check `request.path` to filter the data.</info>
+       * @event Client#apiResponse
+       * @param {APIRequest} request The request that triggered this response
+       * @param {Response} response The response received from the discord API
+       */
+      this.manager.client.emit(API_RESPONSE, request, res);
```

Comment:
I think we shouldn't emit the response we get but a clone of it (res.clone()), to allow users to tap into the raw data without breaking our code due to the data being read already

Reply:
Agh, yes yes, I totally forgot that was an option, and its exactly what I was looking for